tools/nsc/main.cpp

```diff
@@ -110,38 +105,54 @@ class ShaderCompiler final : public system::IApplicationFramework
 
 private:
 
-	core::smart_refctd_ptr<ICPUShader> compile_shader(std::string& shader_code, std::string_view sourceIdentifier) {
+	core::smart_refctd_ptr<ICPUShader> compile_shader(const ICPUShader* shader_code, std::string_view sourceIdentifier) {
 		constexpr uint32_t WorkgroupSize = 256;
 		constexpr uint32_t WorkgroupCount = 2048;
 		const string WorkgroupSizeAsStr = std::to_string(WorkgroupSize);
 
 		smart_refctd_ptr<CHLSLCompiler> hlslcompiler = make_smart_refctd_ptr<CHLSLCompiler>(smart_refctd_ptr(m_system));
 
 		CHLSLCompiler::SOptions options = {};
-		options.stage = asset::IShader::E_SHADER_STAGE::ESS_UNKNOWN;
+		options.stage = shader_code->getStage();
 		//options.debugInfoFlags = IShaderCompiler::E_DEBUG_INFO_FLAGS::EDIF_LINE_BIT;
 		options.preprocessorOptions.sourceIdentifier = sourceIdentifier;
 		options.preprocessorOptions.logger = m_logger.get();
 		options.dxcOptions = std::span<std::string>(m_arguments);
 		auto includeFinder = make_smart_refctd_ptr<IShaderCompiler::CIncludeFinder>(smart_refctd_ptr(m_system));
 		options.preprocessorOptions.includeFinder = includeFinder.get();
 
-		return hlslcompiler->compileToSPIRV(shader_code.c_str(), options);
+		return hlslcompiler->compileToSPIRV((const char*)shader_code->getContent()->getPointer(), options);
 	}
 
 
-	std::string open_shader_file(std::string& filepath) {
-		std::ifstream stream(filepath);
-		std::string str((std::istreambuf_iterator<char>(stream)),
-			std::istreambuf_iterator<char>());
-		return str;
+	const ICPUShader* open_shader_file(std::string& filepath) {
+
+		m_assetMgr = make_smart_refctd_ptr<asset::IAssetManager>(smart_refctd_ptr(m_system));
+		auto resourceArchive = make_smart_refctd_ptr<system::CMountDirectoryArchive>(localInputCWD, smart_refctd_ptr(m_logger), m_system.get());
+		m_system->mount(std::move(resourceArchive));
```

Comment:
should have just set `lp.workingDirectory` to CWD instead of mounting a directory XD

Reply:
I think I did that already, check and if true resolve